plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -134,18 +134,18 @@ public PluginManager(Config cfg) {
     }
 
     private String getUserAgentInformation() {
-        String userAgentInformation= "JenkinsPluginManager";
+        String addonUserAgent = System.getProperty("http.agent");
```

Comment:
Just overwrite this yourself and take full control rather than appending the default one.

Reply:
I would think to be completely backwards compatible that we would still have to include the text 'JenkinsPluginManager' as a default regardless if an http.agent was or was not being provided.  That way it would not break existing implementations, where the Corp proxy, is currently configured to look for that user agent to allow it through.
That follows the pattern of the JVM doing the same

See:
**http.agent (default: “Java/< version >”)**
Defines the string sent in the User-Agent request header in http requests. 
Note that the string “Java/<version>” will be appended to the one provided in the property (e.g. if -Dhttp.agent=”foobar” is used, the User-Agent header will contain “foobar Java/1.5.0” if the version of the VM is 1.5.0). This property is checked only once at startup.
https://docs.oracle.com/javase/8/docs/api/java/net/doc-files/net-properties.html

Reply:
Ok, can you only add the space of the property is set please

Reply:
ok, did that.  Please validate.